src/data/CassandraBackend.h

```diff
@@ -441,6 +441,96 @@ class BasicCassandraBackend : public BackendInterface {
         return {txns, {}};
     }
 
+    NFTsAndCursor
+    fetchNFTsByIssuer(
+        ripple::AccountID const& issuer,
+        std::optional<std::uint32_t> const& taxon,
+        std::uint32_t const ledgerSequence,
+        std::uint32_t const limit,
+        std::optional<ripple::uint256> const& cursorIn,
+        boost::asio::yield_context yield
+    ) const override
+    {
+        NFTsAndCursor ret;
+
+        Statement idQueryStatement = [&taxon, &issuer, &cursorIn, &limit, this]() {
+            if (taxon.has_value()) {
+                auto r = schema_->selectNFTIDsByIssuerTaxon.bind(issuer);
+                r.bindAt(1, *taxon);
+                r.bindAt(2, cursorIn.value_or(ripple::uint256(0)));
+                r.bindAt(3, Limit{limit});
+                return r;
+            }
+
+            auto r = schema_->selectNFTIDsByIssuer.bind(issuer);
+            r.bindAt(
+                1,
+                std::make_tuple(
+                    cursorIn.has_value() ? ripple::nft::toUInt32(ripple::nft::getTaxon(*cursorIn)) : 0,
+                    cursorIn.value_or(ripple::uint256(0))
+                )
+            );
+            r.bindAt(2, Limit{limit});
+            return r;
+        }();
+
+        // Query for all the NFTs issued by the account, potentially filtered by the taxon
+        auto const res = executor_.read(yield, idQueryStatement);
+
+        auto const& idQueryResults = res.value();
+        if (not idQueryResults.hasRows()) {
+            LOG(log_.debug()) << "No rows returned";
+            return {};
+        }
+
+        std::vector<ripple::uint256> nftIDs;
+        for (auto const [nftID] : extract<ripple::uint256>(idQueryResults))
+            nftIDs.push_back(nftID);
+
+        if (nftIDs.size() == 0)
+            return ret;
+
+        if (nftIDs.size() == limit)
+            ret.cursor = nftIDs.back();
+
+        auto nftQueryStatement = schema_->selectNFTBulk.bind(nftIDs);
+        nftQueryStatement.bindAt(1, ledgerSequence);
+
+        // Fetch all the NFT data, meanwhile filtering out the NFTs that are not within the ledger range
+        auto const nftRes = executor_.read(yield, nftQueryStatement);
+        auto const& nftQueryResults = nftRes.value();
+
+        if (not nftQueryResults.hasRows()) {
+            LOG(log_.debug()) << "No rows returned";
```

Comment:
I am thinking maybe we should log error and the nft id here. Is it an error if "nf_tokens" does not contain the corresponding nftIDs nft?

Reply:
> Is it an error if "nf_tokens" does not contain the corresponding nftIDs nft?

Yes, it would be an error is nf_tokens does not contains the corresponding nftIDs, but that doesn't apply in this use case.

No row returned from `nftQueryResults`  isn't necessarily an error. The query `selectNFTBulk` does a _filter_ by ledger sequence on the  `nftIDs` vector that is passed in, so it's completely possibly that every entry in the vector is filtered out, and as a result, the query result is empty.